src/pepr/operator/controllers/istio/istio-resources.ts

```diff
@@ -62,9 +89,112 @@ export async function istioResources(pkg: UDSPackage, namespace: string) {
     serviceEntryNames.set(sePayload.metadata!.name!, true);
   }
 
+  // Reconcile any egress requested
+  await istioEgressResources(
+    createHostResourceMap(pkg),
+    allowList,
+    getPackageId(pkg),
+    pkgName,
+    namespace,
+    generation,
+    ownerRefs,
+  );
+
+  // Purge any orphaned resources
   await purgeOrphans(generation, namespace, pkgName, IstioVirtualService, log);
-  await purgeOrphans(generation, namespace, pkgName, IstioServiceEntry, log);
+  await purgeOrphans(generation, namespace, pkgName, IstioServiceEntry, log); // for ingress and egress
+  await purgeOrphans(generation, namespace, pkgName, IstioSidecar, log); // for egress only
 
   // Return the list of unique hostnames
   return [...hosts];
 }
+
+/**
+ * Creates a ServiceEntry and Sidecar resource for egress traffic and reconciles
+ * shared egress resources
+ *
+ * @param hostResourceMap
+ * @param pkg
+ * @param pkgName
+ * @param namespace
+ * @param generation
+ * @param ownerRefs
+ */
+export async function istioEgressResources(
+  hostResourceMap: HostResourceMap | undefined,
+  allowList: Allow[],
+  pkgId: string,
+  pkgName: string,
+  namespace: string,
+  generation: string,
+  ownerRefs: V1OwnerReference[],
+) {
+  // Error if egress gateway is NOT enabled in the cluster and we have egress resources
+  await K8s(kind.Namespace)
+    .Get(istioEgressGatewayNamespace)
+    .catch(e => {
+      if (hostResourceMap) {
+        if (e.status == 404) {
+          const errText = `Egress gateway is not enabled in the cluster. Please enable the egress gateway and retry.`;
+          log.error(errText);
+          throw new Error(errText);
+        } else {
+          log.error(e, `Unable to reconcile shared egress resources.`);
```

Comment:
slight nit, this is still an error caught when getting the namespace - so maybe something like this.

```suggestion
          log.error(e, `Unable to reconcile get the egress gateway namespace ${istioEgressGatewayNamespace}.`);
```